### PR TITLE
Dps management

### PR DESCRIPTION
**Pull request by EmmaRenauld**
Started adding tests

- Changed dps from an arg in the script (`--dps_keys`) to a line in the config file. This works.
- Copied the file `streamline_containers` from Jeremi's PR and started testing. Does not seem to work yet. Need to check the data shape. Work to be continued.

### CODE REVIEW

**Comment by levje** [comment]
Not to forget: maybe worth adding tests for 2D data_per_streamline arrays before merging.

**Comment by levje** [comment]
Sidenote: with DPS management, the test HDF5 will have the following structure: 
image

File: dwi_ml/data/dataset/streamline_containers.py [hunk 1]
```diff
@@ -8,6 +8,7 @@
 import h5py
 from nibabel.streamlines import ArraySequence
 import numpy as np
+from collections import defaultdict
 
 
 def _load_space_attributes_from_hdf(hdf_group: h5py.Group):
```
Comment (EmmaRenauld) [inline dwi_ml/data/dataset/streamline_containers.py:14]: Could you rename to `_load_streamlines_attributes_from_hdf`?

**Review by EmmaRenauld — COMMENTED** [review]
Looks ok. Did you manage to test with 2D data? I would do that before merging.

**Comment by levje** [comment]
I did manage to get some tests in the PR in test_dataset.py to make sure the mock_2d_dps is the same as expected. I caught a few errors which those tests which are already fixed. Should I write more tests for 2d data?

**Comment by EmmaRenauld** [comment]
No it's ok for now! Let's go forward! Merging

**Pull request merged**